Source/Protobuf/Extensions.cs

```diff
@@ -16,113 +18,94 @@ namespace Dolittle.Protobuf
-            {
-                Key = _.Name,
-                Value = _.Value,
-                ValueType = _.ValueType
-            });
-        }
+        /// <param name="failureReason">The reason for failure.</param>
+        /// <returns>Converted <see cref="grpc.Failure" />.</returns>
+        public static grpc.Failure ToProtobufFailure(this string failureReason) => failureReason.ToProtobufFailure(Guid.Empty);
 
         /// <summary>
-        /// Convert from <see cref="IEnumerable{T}"/> of <see cref="Security.Contracts.Claim"/> to <see cref="Claims"/>.
+        /// Convert a <see cref="string" /> to <see cref="grpc.Failure" /> with given <see cref="Guid" /> as failure id.
         /// </summary>
-        /// <param name="claims"><see cref="IEnumerable{T}"/> of <see cref="Security.Contracts.Claim"/> to convert from.</param>
-        /// <returns>Converted <see cref="Claims"/>.</returns>
-        public static Claims ToClaims(this IEnumerable<Security.Contracts.Claim> claims)
-        {
-            return new Claims(claims.Select(_ => new Claim(_.Key, _.Value, _.ValueType)));
-        }
+        /// <param name="failureReason">The reason for failure.</param>
+        /// <param name="failureId">The failure id.</param>
+        /// <returns>Converted <see cref="grpc.Failure" />.</returns>
+        public static grpc.Failure ToProtobufFailure(this string failureReason, Guid failureId) =>
+            new grpc.Failure { Reason = failureReason, Id = failureId.ToProtobuf() };
+
+        /// <summary>
+        /// Convert a <see cref="Failure" /> to <see cref="grpc.Failure" />.
+        /// </summary>
+        /// <param name="failure"><see cref="Failure" /> to convert.</param>
+        /// <returns>Converted <see cref="grpc.Failure" />.</returns>
+        public static grpc.Failure ToProtobuf(this Failure failure) =>
+            new grpc.Failure { Id = failure.Id.ToProtobuf(), Reason = failure.Reason };
+
+        /// <summary>
+        /// Convert a <see cref="grpc.Failure" /> to <see cref="Failure" />.
+        /// </summary>
+        /// <param name="failure"><see cref="grpc.Failure" /> to convert.</param>
+        /// <returns>Converted <see cref="Failure" />.</returns>
+        public static Failure ToFailure(this grpc.Failure failure) =>
+            new Failure(failure.Id.To<FailureId>(), failure.Reason);
 
         /// <summary>
         /// Convert to metadata that is used as arguments in the header.
         /// </summary>
         /// <param name="message"><see cref="IMessage"/> to convert.</param>
         /// <returns>A <see cref="Metadata.Entry"/> that can be used directly.</returns>
-        public static Metadata.Entry ToArgumentsMetadata(this IMessage message)
-        {
-            return new Metadata.Entry(_argumentsKey, message.ToByteArray());
-        }
+        public static Metadata.Entry ToArgumentsMetadata(this IMessage message) =>
+            new Metadata.Entry(_argumentsKey, message.ToByteArray());
```

Comment:
We shouldn't use this anymore - so get rid of it as well.

Reply:
👍